### PR TITLE
Upgrade laminas-coding-standard to 2.3

### PR DESCRIPTION
**Pull request by arueckauer**
|    Q          |   A
|-------------- | ------
| QA            | yes

### Description

This PR updates the `laminas-coding-standard` to version `2.3.0` and provides necessary code changes.

### CODE REVIEW

File: src/Adapter/Http.php [hunk 1]
```diff
@@ -483,6 +503,10 @@ protected function _basicAuth($header)
 
         // Decode the Authorization header
         $auth = substr($header, strlen('Basic '));
+        if ($auth === false) {
+            return $this->challengeClient();
```
Comment (Ocramius) [inline src/Adapter/Http.php:507]: Checking if a test exists is needed here

**Review by Ocramius — APPROVED** [review]
LGTM - only question is whether the newly introduced conditional requires an additional test

**Comment by Ocramius** [comment]
I will need to create `2.11.x` before merging this

File: src/Adapter/Http.php (hunk 1 above)
Comment (arueckauer) [reply src/Adapter/Http.php:507]: Path was not covered yet. Test added.

File: src/Adapter/DbTable/CredentialTreatmentAdapter.php [hunk 2]
```diff
@@ -107,7 +104,17 @@ protected function authenticateCreateSelect()
      */
     protected function authenticateValidateResult($resultIdentity)
     {
-        if ($resultIdentity['laminas_auth_credential_match'] != '1') {
+        /**
+         * Starting with PHP 8.1.0 integers and floats in result sets will be returned using native PHP types instead
+         * of strings when using emulated prepared statements. To keep the behavior consistent with older versions of
+         * PHP, strict comparison of both types is used.
+         *
+         * @linkhttps://www.php.net/manual/en/migration81.incompatible.php#migration81.incompatible.pdo.mysql
```
Comment (froschdesign) [inline src/Adapter/DbTable/CredentialTreatmentAdapter.php]: A space is missing and the language code in the URL can be omitted; the website automatically redirects.

```suggestion
         * @link https://www.php.net/manual/migration81.incompatible.php#migration81.incompatible.pdo.mysql
```

**Review by Ocramius — APPROVED** [review]
🚢 , thanks @arueckauer!

**Pull request merged**